extensions/classifiers/LDAStringClassifier/LDAStringClassifier.py

```diff
@@ -597,3 +598,91 @@ def predict(self, predicting_data):
         """
         doc_ids = self._add_docs_for_predicting(predicting_data)
         return [self._predict_label_for_doc(doc_id) for doc_id in doc_ids]
+
+    def validate(self, classifier_data):
+        """Validates cached classifier data.
+
+        Args:
+            classifier_data: dict. Consists of various parameter of the
+                classifier.
+        """
+
+        if not isinstance(classifier_data['_alpha'], float):
+            raise utils.ValidationError(
+                ('Expected alpha to be a float, received %f' %
+                 classifier_data['_alpha']))
+
+        if not isinstance(classifier_data['_beta'], float):
+            raise utils.ValidationError(
+                ('Expected beta to be a float, received %f' %
+                 classifier_data['_beta']))
+
+        if not isinstance(classifier_data['_prediction_threshold'], float):
+            raise utils.ValidationError(
+                ('Expected prediction_threshold to be a float, received %f' %
+                 classifier_data['_prediction_threshold']))
+
+        if not isinstance(classifier_data['_training_iterations'], int):
+            raise utils.ValidationError(
+                ('Expected training_iterations to be a int, received %d' %
+                 classifier_data['_training_iterations']))
+
+        if not isinstance(classifier_data['_prediction_iterations'], int):
+            raise utils.ValidationError(
+                ('Expected prediction_iterations to be a int, received %d' %
+                 classifier_data['_prediction_iterations']))
+
+        if not isinstance(classifier_data['_num_labels'], int):
+            raise utils.ValidationError(
+                ('Expected num_labels to be a int, received %d' %
+                 classifier_data['_num_labels']))
+
+        if not isinstance(classifier_data['_num_docs'], int):
+            raise utils.ValidationError(
+                ('Expected num_docs to be a int, received %d' %
+                 classifier_data['_num_docs']))
+
+        if not isinstance(classifier_data['_num_words'], int):
+            raise utils.ValidationError(
+                ('Expected num_words to be a int, received %d' %
+                 classifier_data['_num_words']))
+
+        if not isinstance(classifier_data['_label_to_id'], dict):
```

Comment:
Should we check the types of the dict values here?

Similarly for the types of the list values, below.